apparmor.d/groups/virt/k3s

```diff
@@ -0,0 +1,189 @@
+# apparmor.d - Full set of apparmor profiles
+# Copyright (C) 2022 Jeroen Rijken
+# SPDX-License-Identifier: GPL-2.0-only
+
+abi <abi/3.0>,
+
+include <tunables/global>
+
+@{exec_path} = /{usr/,}{local/,}bin/k3s
+profile k3s @{exec_path} flags=(complain) {
+  include <abstractions/base>
+  include <abstractions/disks-read>
+  include <abstractions/nameservice-strict>
+  include <abstractions/ssl_certs>
+
+  capability chown,
+  capability dac_override,
+  capability dac_read_search,
+  capability net_admin,
+  capability syslog,
+  capability sys_admin,
+  capability sys_resource,
+
+  ptrace peer=@{profile_name},
+  ptrace (read) peer=unconfined,
+
+  network inet dgram,
+  network inet6 dgram,
+  network inet stream,
+  network inet6 stream,
+  network netlink raw,
+
+  mount /var/lib/kubelet/pods/@{uuid}/volumes/kubernetes.io~*/{,**/},
+  umount /var/lib/kubelet/pods/@{uuid}/volumes/kubernetes.io~*/{,**/},
+
+  signal (send, receive) set=term,
+
+  @{exec_path} mr,
+  /{usr/,}bin/kmod rPx,
+  /{usr/,}bin/mount rPx,
+  /{usr/,}bin/systemd-run rix,
+  /{usr/,}{s,}bin/xtables-nft-multi rCx -> xtables-nft-multi,
+
+  @{libexec}/kubernetes/kubelet-plugins/volume/exec/nodeagent~uds/uds rix,
+  /var/lib/rancher/k3s/data/[0-9a-f]*/bin/* rix,
+
+  @{libexec}/kubernetes/kubelet-plugins/volume/exec/{,**} r,
+  /usr/share/mime/globs2 r,
+
+  /etc/machine-id r,
+  /etc/rancher/k3s/{,**} r,
+  /etc/rancher/k3s/k3s.yaml rw,
+  /etc/rancher/node/password r,
+
+  /var/lib/rancher/k3s/{,**} r,
+  /var/lib/rancher/k3s/agent/** rw,
+  /var/lib/rancher/k3s/server/** rw,
+  /var/lib/rancher/k3s/server/db/** rwk,
+
+  # k3s want's to basically manage all directories and create some specific files.
+  /var/lib/kubelet/{,**/} rw,
+  /var/lib/kubelet/{cpu_manager_state,memory_manager_state} r,
+  /var/lib/kubelet/device-plugins/{,DEPRECATION,kubelet.sock} rw,
+  /var/lib/kubelet/pod-resources/{kubelet.sock,[0-9]*} rw,
+  /var/lib/kubelet/pods/@{uuid}/containers/*/[0-9a-f]* rw,
+  /var/lib/kubelet/pods/@{uuid}/etc-hosts rw,
+  /var/lib/kubelet/pods/@{uuid}/plugins/kubernetes.io~*/{,**} rw,
+  /var/lib/kubelet/pods/@{uuid}/volumes/kubernetes.io~*/{,**} rw,
+  /var/lib/kubelet/pods/@{uuid}/**/ca.crt rw,
+  /var/lib/kubelet/pods/@{uuid}/**/namespace rw,
+  /var/lib/kubelet/pods/@{uuid}/**/token rw,
+
+  /var/log/containers/ r,
+  /var/log/containers/** rw,
+  /var/log/rancher/{,**} r,
+  /var/log/kubelet/{,**} r,
+  /var/log/kubernetes/{,**} r,
+  /var/log/kubernetes/audit/** rw,
+  /var/log/pods/{,**} r,
+  /var/log/pods/{,**/} rw,
+  /var/log/pods/**/[0-9]*.log rw,
+
+  owner @{HOME}/.kube/cache/discovery/{,**} rw,
+  owner @{HOME}/.kube/cache/http/[0-9a-z]* rw,
+  owner @{HOME}/.kube/cache/http/.diskv-temp/[0-9]* rw,
```

Comment:
Same here, why not `owner @{HOME}/.kube/cache/{,**} rw,`?

Reply:
This could be simplified indeed.